packages/wallets/src/components/WalletListCardDropdown/WalletListCardDropdown.tsx

```diff
@@ -27,66 +36,107 @@ const WalletListCardDropdown: React.FC<TSubscribedBalance> = ({ balance }) => {
         [t]
     );
 
-    useEffect(() => {
-        const selectedWallet = wallets?.find(wallet => wallet.loginid === loginId);
-        if (selectedWallet) {
-            const selectedTextWidth = generateTitleText(selectedWallet).length;
-            setInputWidth(`${selectedTextWidth * 10 - 20}px`);
-        }
-    }, [generateTitleText, wallets, loginId]);
-
     const walletList: WalletList = useMemo(() => {
-        return wallets
-            ?.filter(wallet => !wallet.is_virtual)
-            ?.map(wallet => ({
-                listItem: (
-                    <div className='wallets-list-card-dropdown__item'>
-                        <WalletCurrencyIcon currency={wallet.currency ?? 'USD'} rounded />
-                        <div className='wallets-list-card-dropdown__item-content'>
-                            <WalletText size='2xs'>
-                                <Trans defaults={`${wallet.currency} Wallet`} />
-                            </WalletText>
-                            <WalletText size='sm' weight='bold'>
-                                <Trans
-                                    defaults={displayMoney?.(
-                                        balanceData?.accounts?.[wallet.loginid]?.balance ?? 0,
-                                        wallet?.currency || '',
-                                        {
-                                            fractional_digits: wallet?.currency_config?.fractional_digits,
-                                        }
-                                    )}
-                                />
-                            </WalletText>
+        return (
+            wallets
+                ?.filter(wallet => !wallet.is_virtual)
+                .map(wallet => ({
+                    listItem: (
+                        <div className='wallets-listcard-dropdown__list-item'>
```

Comment:
There are still nodes here. There are better ways to structure this. I will swing by.